applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py

```diff
@@ -0,0 +1,682 @@
+
+            self.analyzer.AnalyzeDesignAndReportToCommunicator(self.design_surface, self.opt_iteration, self.communicator)
+
+    # --------------------------------------------------------------------------
+    def __PostProcessGradientsObtainedFromAnalysis(self):
+        self.geometry_utilities.ComputeUnitSurfaceNormals()
+
+        # Process objective gradients
+        obj = self.specified_objectives[0]
+        obj_id = obj["identifier"].GetString()
+
+        obj_gradients_dict = self.communicator.getStandardizedGradient(obj_id)
+
+        nodal_variable = KratosGlobals.GetVariable("DF1DX")
+        WriteDictionaryDataOnNodalVariable(obj_gradients_dict, self.optimization_model_part, nodal_variable)
+
+        # Projection on surface normals
+        if obj["project_gradient_on_surface_normals"].GetBool():
+            self.geometry_utilities.ProjectNodalVariableOnUnitSurfaceNormals(nodal_variable)
+
+        # Damping
+        if self.is_damping_specified:
+            self.damping_utilities.DampNodalVariable(nodal_variable)
+
+        # Mapping
+        nodal_variable_mapped = KratosGlobals.GetVariable("DF1DX_MAPPED")
+        self.mapper.MapToDesignSpace(nodal_variable, nodal_variable_mapped)
+        self.mapper.MapToGeometrySpace(nodal_variable_mapped, nodal_variable_mapped)
+
+        # Damping
+        if self.is_damping_specified:
+            self.damping_utilities.DampNodalVariable(nodal_variable_mapped)
+
+        # Process constraint gradients
+        for itr in range(self.specified_constraints.size()):
+            con = self.specified_constraints[itr]
+
+            # Process equality constraints
+            if con["type"].GetString()=="=":
+                eq_id = con["identifier"].GetString()
+
+                eq_gradients_dict = self.communicator.getStandardizedGradient(eq_id)
+
+                nodal_variable = KratosGlobals.GetVariable("DC"+str(itr+1)+"DX")
+                WriteDictionaryDataOnNodalVariable(eq_gradients_dict, self.optimization_model_part, nodal_variable)
+
+                # Projection on surface normals
+                if con["project_gradient_on_surface_normals"].GetBool():
+                    self.geometry_utilities.ProjectNodalVariableOnUnitSurfaceNormals(nodal_variable)
+
+                # Damping
+                if self.is_damping_specified:
+                    self.damping_utilities.DampNodalVariable(nodal_variable)
+
+                # Mapping
+                nodal_variable_mapped = KratosGlobals.GetVariable("DC"+str(itr+1)+"DX_MAPPED")
+                self.mapper.MapToDesignSpace(nodal_variable, nodal_variable_mapped)
+                self.mapper.MapToGeometrySpace(nodal_variable_mapped, nodal_variable_mapped)
+
+                # Damping
+                if self.is_damping_specified:
+                    self.damping_utilities.DampNodalVariable(nodal_variable_mapped)
+
+            # Process inequality constraints
+            else:
```

Comment:
also here the same is happening for eq and ineq

Reply:
True. Fixed it.